### PR TITLE
feat: Add the Cluster Autoscaler

### PR DESCRIPTION
**Pull request by aakilin**
- fix sync-wave inside the application folder
- add IAM module "cluster_autoscaler_irsa_role"
- move the cluster-autoscaler path to kube-system
- inner wave shouldn’t be 30th - make sense to set it 0
- clustername inside the values should be a tokenized by `<PRIMARY_CLUSTER_NAME>`
- aws-region should be tokenized by `<CLOUD_REGION>`
- cloud-provider should be tokenized by `<CLOUD_PROVIDER>`
- service-account should be tokenized by SERVICE_ACCOUNT - create the new one
`<K8S_ROLE_MAPPING>: “<CLUSTER_AUTOSCALER_IAM_ROLE_RN>”`
- add the output cluster_autoscaler_role with `value       = module.hosting-provider.cluster_autoscaler_irsa_role`
- add `p.parameters["<CLUSTER_AUTOSCALER_IAM_ROLE_RN>"] = hp_out["cluster_autoscaler_role"]`
- add the `cluster_autoscaler_irsa_role` to the terraform `cloud_aws` module outputs

### CODE REVIEW

File: tools/cli/services/cloud/aws/aws_manager.py [hunk 1]
```diff
@@ -171,5 +171,6 @@ def create_external_secrets_config(self, **kwargs) -> str:
         return ""
 
     @trace()
-    def create_autoscaler_snippet(self, cluster_name: str, node_groups=[]):
-        return '''awsRegion: <CLOUD_REGION>'''
+    def create_iac_pr_automation_config_snippet(self):
```
Comment (sergs-pci) [inline tools/cli/services/cloud/aws/aws_manager.py:174]: Looks wrong

File: tools/cli/services/cloud/azure/azure_manager.py [hunk 2]
```diff
@@ -236,28 +236,14 @@ def create_external_secrets_config(self, **kwargs) -> str:
                 "subscriptionId": "{subscription_id}",
                 "resourceGroup": "{resource_group}",
                 "useWorkloadIdentityExtension": true
-              }}'''.format(subscription_id=self.__azure_sdk.subscription_id, resource_group=location)
+              }}'''.format(subscription_id=self._azure_sdk.subscription_id, resource_group=location)
 
     @trace()
-    def create_autoscaler_snippet(self, cluster_name: str, node_groups=[]):
```
Comment (sergs-pci) [inline tools/cli/services/cloud/azure/azure_manager.py]: the same as above, looks like with this conflict resolving you are deleting necessary changes

**Review by sergs-pci — CHANGES_REQUESTED** [review]

File: tools/cli/services/cloud/azure/azure_manager.py [hunk 3]
```diff
@@ -238,6 +238,30 @@ def create_external_secrets_config(self, **kwargs) -> str:
...
+        return f'''autoscalingGroups: {autoscaling_groups}
+        azureClientID: "<CLUSTER_AUTOSCALER_IAM_ROLE_RN>"
+        azureResourceGroup: {cluster_name}-vmss-rg
+        azureSubscriptionID: {self.__azure_sdk.subscription_id}
+        azureTenantID: {tenant_id}
+        azureUseWorkloadIdentityExtension: true
+        azureVMType: "vmss"'''
+
     @trace()
     def create_iac_pr_automation_config_snippet(self):
```
Comment (sergs-pci) [inline tools/cli/services/cloud/azure/azure_manager.py:266]: looks wrong that you remain these lines

**Review by sergs-pci — CHANGES_REQUESTED** [review]

**Review by sergs-pci — DISMISSED** [review]

File: tools/cli/services/cloud/azure/azure_manager.py [hunk 4]
```diff
@@ -20,11 +20,11 @@ def __init__(
             self, subscription_id: str, location: Optional[str] = None, storage_container_name: Optional[str] = None
     ):
         self.iac_backend_storage_container_name: Optional[str] = storage_container_name
-        self._azure_sdk = AzureSdk(subscription_id, location)
+        self.__azure_sdk = AzureSdk(subscription_id, location)
```
Comment (all4code) [inline tools/cli/services/cloud/azure/azure_manager.py:23]: Should use name with one underscore.

**Review by all4code — CHANGES_REQUESTED** [review]

**Review by all4code — APPROVED** [review]

**Review by sergs-pci — APPROVED** [review]

**Pull request merged**